### ISSUE TITLE
regression in: npm shrinkwrap install from local tarball

### ISSUE THREAD

**Issue by hoskeri** [npm/npm#13030]
Installing from shrinkwrap with resolved urls that are relative to the package root used to resolve the path to the tarball relative to the package root. 

Somewhere in the 3.9 series, this seems to have changed, and the paths are now resolved relative to the root of its *dependency*

For example, given the shrinkwrapped dependency:
```
"semver": {
          "version": "5.0.3",
          "from": "semver@>=5.0.1 <5.1.0",
          "resolved": "./node_shrinkwrap/semver-5.0.3.tgz"
        }
```

install will attempt to look for `$package_root/node_modules/${dependency}/semver-5.0.3.tgz`

If this is not a regression, then what is the correct directory to specify relative tarballs to?

**Comment by hoskeri** [comment]
seems to have been broken/changed with 3.8.8 release

https://github.com/npm/npm/commit/05a28e38586082ac4bbf26ee6f863cc8d07054d6

and/or https://github.com/npm/npm/commit/ada2e93e8b276000150a9aa93fff69ec366e03d6

**Comment by iarna** [comment]
This should be fixed as of 3.10.0. Please give the latest `npm` a try! I'm going to close this– if you find you're still having issues please open a new one. Thanks!
